Entity/Dynamic.php

```diff
@@ -2,118 +2,288 @@
+    const TYPE_HEADLINE = 'headline';
+    const TYPE_ATTACHMENT = 'attachment';
+    const TYPE_CHECKBOX = 'checkbox';
+    const TYPE_CHECKBOX_MULTIPLE = 'checkboxMultiple';
+    const TYPE_DROPDOWN = 'dropdown';
+    const TYPE_DROPDOWN_MULTIPLE = 'dropdownMultiple';
+    const TYPE_RADIO_BUTTONS = 'radioButtons';
+
+    /**
+     * @var int
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    private $uuid;
+
+    /**
+     * @var string
+     */
+    private $locale;
+
+    /**
+     * @var int
+     */
+    private $formId;
+
+    /**
+     * @var string
+     */
+    private $webspaceKey;
+
+    /**
+     * @var string
+     */
+    private $salutation;
+
+    /**
+     * @var string
+     */
+    private $title;
+
+    /**
+     * @var string
+     */
+    private $firstName;
+
+    /**
+     * @var string
+     */
+    private $lastName;
+
+    /**
+     * @var string
+     */
+    private $email;
+
+    /**
+     * @var string
+     */
+    private $phone;
+
+    /**
+     * @var string
+     */
+    private $fax;
+
+    /**
+     * @var string
+     */
+    private $street;
+
+    /**
+     * @var string
+     */
+    private $zip;
+
+    /**
+     * @var string
+     */
+    private $city;
+
+    /**
+     * @var string
+     */
+    private $state;
+
     /**
      * @var string
      */
-    private $form;
+    private $country;
 
     /**
      * @var string
      */
-    private $data;
+    private $function;
+
+    /**
+     * @var string
+     */
+    private $company;
+
+    /**
+     * @var string
+     */
+    private $text;
+
+    /**
+     * @var string
+     */
+    private $textarea;
 
     /**
      * @var \DateTime
      */
-    private $created;
+    private $date;
 
     /**
-     * @var int
+     * @var string
      */
-    private $id;
+    private $attachment;
 
     /**
-     * set created time.
+     * @var string
      */
-    public function __construct()
-    {
-        $this->created = new \DateTime();
-    }
+    private $checkbox;
 
     /**
-     * Set form.
-     *
-     * @param string $form
-     *
-     * @return Dynamic
+     * @var string
      */
-    public function setForm($form)
-    {
-        $this->form = $form;
+    private $checkboxMultiple;
 
-        return $this;
-    }
+    /**
+     * @var string
+     */
+    private $dropdown;
+
+    /**
+     * @var string
+     */
+    private $dropdownMultiple;
+
+    /**
+     * @var string
+     */
+    private $radioButtons;
 
     /**
-     * Get form.
+     * @var string
+     */
+    private $data = '[]';
+
+    /**
+     * @var \DateTime
+     */
+    private $created;
+
+    /**
+     * @var \DateTime
+     */
+    private $changed;
+
+    /**
+     * Returns an array of all constants.
      *
-     * @return string
+     * @return array
      */
-    public function getForm()
+    public static function getConstants()
     {
-        return $this->form;
+        $reflectionClass = new ReflectionClass(__CLASS__);
```

Comment:
@thomasduenser avoid reflection as it is a performance bottleneck. create a static variable.

```php
public static $TYPES = [
    self::TYPE_SPACER,
    self::TYPE_FREE_TEXT,
    self::TYPE_SALUTATION,
    self::TYPE_TITLE,
    self::TYPE_FIRST_NAME,
    self::TYPE_LAST_NAME,
    self::TYPE_EMAIL,
    self::TYPE_PHONE,
    self::TYPE_FAX,
    self::TYPE_STREET,
    self::TYPE_ZIP,
    self::TYPE_CITY,
    self::TYPE_STATE,
    self::TYPE_COUNTRY,
    self::TYPE_FUNCTION,
    self::TYPE_COMPANY,
    self::TYPE_TEXT,
    self::TYPE_TEXTAREA,
    self::TYPE_DATE,
    self::TYPE_HEADLINE,
    self::TYPE_ATTACHMENT,
    self::TYPE_CHECKBOX,
    self::TYPE_CHECKBOX_MULTIPLE,
    self::TYPE_DROPDOWN,
    self::TYPE_DROPDOWN_MULTIPLE,
    self::TYPE_RADIO_BUTTONS,
];
```